### ISSUE TITLE
Auto-generate docs for sprint data-rest?

### ISSUE THREAD

**Issue by bedge** [fabiomaffioletti/jsondoc#112]
We have some explicit controllers for which jsondoc is great, but we also have some completely generated ones from spring data-rest for which there's no explicit source code.
Is there any global switch one can flip to enable jsondoc to generate docs for the data-rest generated controllers as well?

**Comment by fabiomaffioletti** [comment]
That would be awesome. Currently there is no configuration to do that, but it's definitely something worth to be added in the future.

**Comment by bedge** [comment]
Is there any mechanism available to document the data-rest generated controllers, automated or not?

**Comment by fabiomaffioletti** [comment]
No, at this moment there is no way to do it in an automatic way. Anyway, you can use a trick: create a "fake documentation class", without any spring annotation, just jsondoc ones, and let jsondoc scan it. There you can have jsondoc annotated methods, reflecting the ones generated by spring data rest. In this way you can still document your api, even if in a non automatic way. Let me know if this is clear enough.

**Comment by bedge** [comment]
When you say "reflecting" are you implying that one can rely on java reflection to automate some aspect of this, or does one need to manually create every generated class/method's jsondoc annotations?

**Comment by fabiomaffioletti** [comment]
No, I mean this: spring data rest generates crud operations for your domain model for example Person, but they are not visible in the code because they are hidden by the framework. "Reflecting" means that to document them you could create a class which only has documentation purposes, name it like  "PersonDocumentation", in which you define the same methods spring data rest would do, and put jsondoc annotations on them. The result will be that your methods (with no business logic inside) will document the real ones auto generated by spring data rest.

**Comment by bedge** [comment]
OK, got it. Ugh, that sounds very manually intensive. I'll keep looking for some other way. I don't mind of the custom controllers and generated ones use different document systems, although uniformity would be preferable. 
I suppose the real solution for this would have to be embedded with the spring code generator in order to generate the annotations as the spring data-rest controllers are generated.

**Comment by fabiomaffioletti** [comment]
Yes, that's correct.
